src/librbd/ImageCtx.cc

```diff
@@ -850,7 +850,11 @@ struct C_InvalidateCache : public Context {
     on_safe = util::create_async_context_callback(*this, on_safe);
 
     assert(owner_lock.is_locked());
-    if (object_cacher != NULL) {
+
+    bool journaling = false;
+    journaling = (journal != NULL && !journal->is_journal_replaying());
+
+    if (!journaling && object_cacher != NULL) {
```

Comment:
@dillaman so in a system flush, we do need to flush the cache even if journaling is enabled. is my understanding right here?